connector/atlassiancrowd/atlassiancrowd.go

```diff
@@ -368,6 +374,15 @@ func (c *crowdConnector) identityFromCrowdUser(user crowdUser) (connector.Identi
 		EmailVerified: true,
 	}
 
+	switch c.PreferredUsernameField {
+	case "key":
+		identity.PreferredUsername = user.Key
+	case "name":
+		identity.PreferredUsername = user.Name
+	case "email":
+		identity.PreferredUsername = user.Email
+	}
```

Comment:
In my opinion, we need to add a warning (or return an error) if the preferredUserName field is set to the wrong value.

Users can start wondering why the claim is empty after setting the option value to 'email or another invalid value.